Sources/TuistPlugin/PluginService.swift

```diff
@@ -316,6 +323,19 @@ public final class PluginService: PluginServicing {
         }
     }
 
+    func getPluginDownloadUrl(gitUrl: String, gitTag: String, pluginName: String, releaseUrl: String?) -> URL? {
+        if let url = releaseUrl.flatMap(URL.init(string:)) {
+            return url
+        }
```

Comment:
Shouldn't we allow to interpolate the tag somehow? Otherwise you need to keep the url in sync with the version

Reply:
Since it is a direct download link I think it is up to the publisher to include tag in the URL

Reply:
The configuration is something like:
```
    .git(
      url: "your repo url",
      tag: "1.2.3",
      releaseURL: "…/1.2.3/…"
    ),
```

You don't want to have to update the releaseURL every time you update the tag

Reply:
Both the `gitUrl` and `gitTag` parameters become useless if you use the `releaseUrl` as they are used to construct that URL. Is there maybe a way we can get the URL for gitlab by constructing it? If not, then developers should not have to specify `url`, `tag`, _and_ `releaseURL`.

Reply:
I think the problem is that if the repo is private the GitHub/GitLab release url doesn't work

Reply:
I see – then in that case we can allow hardcording the release url and skip the `url` and `tag` parameters as they are useless at that point afaik.

Reply:
nope, they are still needed to download the "ProjectDescriptionHelper" part of the plugin via git

Reply:
ahh, I see, then I think it's gonna be more straightforward not to use interpolation here.